File: operaciones_matrices.py
# Función para ingresar los valores de una matriz
def completar_matrices(nombre):
    matriz = []

    print(f"\nIngrese los valores de la matriz {nombre}:")

    for fila in range(2):
        nueva_fila = []

        for columna in range(3):
            while True:
                try:
                    valor = int(
                        input(f"{nombre}[{fila}][{columna}]: ")
                    )
                    nueva_fila.append(valor)
                    break
                except ValueError:
                    print("Error: debe ingresar un número entero.")

        matriz.append(nueva_fila)

    return matriz


# Función para sumar dos matrices
def suma_matriz(matriz_a, matriz_b):
    resultado = [[0, 0, 0], [0, 0, 0]]

    for fila in range(2):
        for columna in range(3):
            resultado[fila][columna] = (
                matriz_a[fila][columna]
                + matriz_b[fila][columna]
            )

    return resultado


# Función para restar dos matrices
def resta_matriz(matriz_a, matriz_b):
    resultado = [[0, 0, 0], [0, 0, 0]]

    for fila in range(2):
        for columna in range(3):
            resultado[fila][columna] = (
                matriz_a[fila][columna]
                - matriz_b[fila][columna]
            )

    return resultado


# Función para multiplicar elemento por elemento
def multiplicacion_matriz(matriz_a, matriz_b):
    resultado = [[0, 0, 0], [0, 0, 0]]

    for fila in range(2):
        for columna in range(3):
            resultado[fila][columna] = (
                matriz_a[fila][columna]
                * matriz_b[fila][columna]
            )

    return resultado


# Función para mostrar una matriz
def mostrar_resultado(matriz):
    for fila in matriz:
        print("|", end=" ")

        for valor in fila:
            print(valor, end=" ")

        print("|")


# Función principal
def main():
    matriz_a = completar_matrices("A")
    matriz_b = completar_matrices("B")

    resultado_suma = suma_matriz(matriz_a, matriz_b)
    resultado_resta = resta_matriz(matriz_a, matriz_b)

    resultado_multiplicacion = multiplicacion_matriz(
        matriz_a,
        matriz_b
    )

    print("\nMatriz A:")
    mostrar_resultado(matriz_a)

    print("\nMatriz B:")
    mostrar_resultado(matriz_b)

    print("\nResultado de la suma:")
    mostrar_resultado(resultado_suma)

    print("\nResultado de la resta:")
    mostrar_resultado(resultado_resta)

    print("\nResultado de la multiplicación:")
    mostrar_resultado(resultado_multiplicacion)


# Iniciar el programa
if __name__ == "__main__":
    main()

File: operacionesMatrices.java
import java.util.Scanner;

public class OperacionesMatrices {

    /*
     * Función para ingresar los valores de una matriz.
     */
    public static void completarMatrices(
        Scanner entrada,
        int[][] matriz,
        String nombre
    ) {
        System.out.println(
            "\nIngrese los valores de la matriz " + nombre + ":"
        );

        for (int fila = 0; fila < 2; fila++) {
            for (int columna = 0; columna < 3; columna++) {
                System.out.print(
                    nombre + "[" + fila + "][" + columna + "]: "
                );

                matriz[fila][columna] = entrada.nextInt();
            }
        }
    }

    /*
     * Función para sumar dos matrices.
     */
    public static void sumaMatriz(
        int[][] matrizA,
        int[][] matrizB,
        int[][] resultado
    ) {
        for (int fila = 0; fila < 2; fila++) {
            for (int columna = 0; columna < 3; columna++) {
                resultado[fila][columna] =
                    matrizA[fila][columna] + matrizB[fila][columna];
            }
        }
    }

    /*
     * Función para restar dos matrices.
     */
    public static void restaMatriz(
        int[][] matrizA,
        int[][] matrizB,
        int[][] resultado
    ) {
        for (int fila = 0; fila < 2; fila++) {
            for (int columna = 0; columna < 3; columna++) {
                resultado[fila][columna] =
                    matrizA[fila][columna] - matrizB[fila][columna];
            }
        }
    }

    /*
     * Función para multiplicar elemento por elemento.
     */
    public static void multiplicacionMatriz(
        int[][] matrizA,
        int[][] matrizB,
        int[][] resultado
    ) {
        for (int fila = 0; fila < 2; fila++) {
            for (int columna = 0; columna < 3; columna++) {
                resultado[fila][columna] =
                    matrizA[fila][columna] * matrizB[fila][columna];
            }
        }
    }

    /*
     * Función para mostrar una matriz.
     */
    public static void mostrarResultado(int[][] matriz) {
        for (int fila = 0; fila < 2; fila++) {
            System.out.print("| ");

            for (int columna = 0; columna < 3; columna++) {
                System.out.print(matriz[fila][columna] + " ");
            }

            System.out.println("|");
        }
    }

    /*
     * Método principal.
     */
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);

        int[][] matrizA = new int[2][3];
        int[][] matrizB = new int[2][3];
        int[][] resultadoSuma = new int[2][3];
        int[][] resultadoResta = new int[2][3];
        int[][] resultadoMultiplicacion = new int[2][3];

        completarMatrices(entrada, matrizA, "A");
        completarMatrices(entrada, matrizB, "B");

        sumaMatriz(matrizA, matrizB, resultadoSuma);
        restaMatriz(matrizA, matrizB, resultadoResta);

        multiplicacionMatriz(
            matrizA,
            matrizB,
            resultadoMultiplicacion
        );

        System.out.println("\nMatriz A:");
        mostrarResultado(matrizA);

        System.out.println("\nMatriz B:");
        mostrarResultado(matrizB);

        System.out.println("\nResultado de la suma:");
        mostrarResultado(resultadoSuma);

        System.out.println("\nResultado de la resta:");
        mostrarResultado(resultadoResta);

        System.out.println("\nResultado de la multiplicación:");
        mostrarResultado(resultadoMultiplicacion);

        entrada.close();
    }
}